### ISSUE TITLE
OOM crash on `./gzemnid.js depsdb`

### ISSUE THREAD

**Issue by davisjam** [nodejs/Gzemnid#16]
```
(16:09:40) jamie@woody ~/Desktop/node-threadpool-use/Gzemnid $ tail -30 /tmp/gzemnid_debsdb
Read 510000...
Read 520000...
Read 530000...
Read 540000...
Read 550000...
Read 560000...
Read 570000...

<--- Last few GCs --->

[16616:0x3643a30]   480457 ms: Mark-sweep 1403.2 (1483.9) -> 1403.2 (1452.9) MB, 1703.3 / 0.0 ms  (+ 0.0 ms in 0 steps since start of marking, biggest step 0.0 ms, walltime since start of marking 1724 ms) last resort GC in old space requested
[16616:0x3643a30]   482227 ms: Mark-sweep 1403.2 (1452.9) -> 1403.2 (1452.9) MB, 1769.0 / 0.0 ms  last resort GC in old space requested


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x28e0da6a5ee1 <JSObject>
    1: /* anonymous */ [/home/jamie/Desktop/node-threadpool-use/Gzemnid/node_modules/jsonparse/jsonparse.js:~342] [pc=0x2093e9cad6f4](this=0x3b7c97682449 <Parser map = 0x1c2a26b5bb71>,token=1,value=0x285bb0f65161 <String[1]: {>)
    2: /* anonymous */ [/home/jamie/Desktop/node-threadpool-use/Gzemnid/node_modules/jsonparse/jsonparse.js:~127] [pc=0x2093e9c9db63](this=0x3b7c97682449 <Parser map = 0...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [node]
 2: 0x121a7ac [node]
 3: v8::Utils::ReportOOMFailure(char const*, bool) [node]
 4: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [node]
 5: v8::internal::Factory::NewFillerObject(int, bool, v8::internal::AllocationSpace) [node]
 6: v8::internal::Runtime_AllocateInTargetSpace(int, v8::internal::Object**, v8::internal::Isolate*) [node]
 7: 0x2093e9b0463d
```

**Comment by davisjam** [comment]
This happens during the `resolved` stage; it happened again when I ran

`./gzemnid.js debsdb resolved`.

just after "Read 560K".

**Comment by davisjam** [comment]
Possibly addressed by `--max-old-space-size` as noted at the top of the README.

**Comment by ChALkeR** [comment]
Yes, it should be run with `--max-old-space-size=2000` or above.
Perhaps this needs a wrapper script that sets that flag and/or a check that prints a warning if specific commands (including `depsdb`) are executed without increased memory limit.

Could you confirm that it works with `--max-old-space-size=2000`?

**Comment by davisjam** [comment]
> Could you confirm that it works with --max-old-space-size=2000?

It works!

> Perhaps this needs a wrapper script that sets that flag and/or a check that prints a warning if specific commands (including depsdb) are executed without increased memory limit.

Good idea.

**Comment by ChALkeR** [comment]
@davisjam 

This is now fixed by d85854a880d5a3acdc6e10df628db5695234ac39 and d85854a880d5a3acdc6e10df628db5695234ac39, which significantly reduce the memory usage of `depsdb` so it now fits in the default limit.

Automatic re-launching with `--max-old-space-size` could be done later if/when required.
